testsuite/tests/typing-layouts/basics.ml

```diff
@@ -0,0 +1,982 @@
+(* TEST
+   * expect
+*)
+type t_any   [@@any]
+type t_value [@@value]
+type t_imm   [@@immediate]
+type t_imm64 [@@immediate64]
+type t_void  [@@void]
+
+type void_variant = VV of t_void
```

Comment:
The restriction is more like "you can't have an empty block".  In particular, constructors with no arguments are already a thing, and are represented as an immediate (which is what you get here).  But records with no fields are not already a thing.  While it would be fun to add them, it's probably beyond the scope of v1.